### PR TITLE
[dynamo] Add support for itertools.repeat

### LINKED ISSUE #110286: [dynamo] support itertools.repeat: call_function repeat in skip_files Builtin repeat, skip reason: filename is None
### 🐛 Describe the bug

self explanatory

we should use the polyfill approach to implement this see also https://github.com/pytorch/pytorch/pull/109284#issuecomment-1737533266 https://github.com/pytorch/pytorch/pull/109972#discussion_r1336174289

Real world use case in torchrec dist_data.py

```
            batch_size_per_feature = list(
                itertools.chain.from_iterable(
                    itertools.repeat(x, local_split) for x in batch_size_per_rank
                )
            )
```

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### PR DESCRIPTION
**Pull request by issamarabi**
Implemented call_repeat function in builtin.py to handle itertools.repeat within the Dynamo context. This addresses issue #110286 where support for itertools.repeat was requested.

Fixes #110286

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by voznesenskym — CHANGES_REQUESTED** [review]
Tests, also, odd that its routing through builtins without a value check of itertools.repeat.

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -1503,6 +1503,18 @@ def call_not_(self, tx, a):
             return ConstantVariable.create(len(a.items) == 0).add_options(self, a)
 
         return None
+    
+    def call_repeat(self, tx, a, b):
+        if isinstance(a, SymNodeVariable) and isinstance(b, ConstantVariable):
+            return SymNodeVariable.create(
```
Comment (voznesenskym) [inline torch/_dynamo/variables/builtin.py:1509]: This is the wrong return type, try it in eager:

image

You need an iterator type.

Comment (jon-chuang) [reply torch/_dynamo/variables/builtin.py:1509]: This is not a builtin, but is handled as a misc var: https://github.com/jon-chuang/pytorch/blob/e1ea8f08fa081e5544bd8326552daf8d47c92d92/torch/_dynamo/variables/misc.py#L793

File: torch/_dynamo/variables/lists.py [hunk 2]
```diff
@@ -729,6 +729,29 @@ class TupleIteratorVariable(ListIteratorVariable):
     pass
 
 
+class RepeatIteratorVariable(VariableTracker):
+    def __init__(self, item, count=None, **kwargs):
```
Comment (jon-chuang) [inline torch/_dynamo/variables/lists.py:733]: count should never be None

**Pull request closed without merging**